charts/newrelic-k8s-metrics-adapter/templates/_helpers.tpl

```diff
@@ -47,3 +47,31 @@ Naming helpers
 {{- define "newrelic-k8s-metrics-adapter.name.hpa-controller" -}}
 {{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "hpa-controller") }}
 {{- end -}}
+
+{{/*
+Return the custom secret name where the NR Personal API key is being stored.
+*/}}
+{{- define "newrelic-k8s-metrics-adapter.customSecretPersonalApiKeyName" -}}
+    {{- .Values.customSecretPersonalApiKeyName | default "" -}}
+{{- end -}}
+
+{{/*
+Return the custom secret key name  where the NR Personal API key is being stored.
+*/}}
+{{- define "newrelic-k8s-metrics-adapter.customSecretPersonalApiKeyKey" -}}
+    {{- .Values.customSecretPersonalApiKeyKey | default "personalAPIKey" -}}
+{{- end -}}
+
+{{- define "newrelic-k8s-metrics-adapter.personalAPIKey" -}}
+    {{- .Values.personalAPIKey | default "" -}}
+{{- end -}}
+
+{{/*
+Returns if the template should render, it checks if the required values
+personalAPIKey or personalAPIKey
+*/}}
+{{- define "newrelic-k8s-metrics-adapter.areValuesValid" -}}
+{{- $personalAPIKey := include "newrelic-k8s-metrics-adapter.personalAPIKey" . -}}
+{{- $customSecretPersonalApiKeyName := include "newrelic-k8s-metrics-adapter.customSecretPersonalApiKeyName" . -}}
+{{- and (or $personalAPIKey $customSecretPersonalApiKeyName) }}
+{{- end -}}
```

Comment:
Instead of doing this in a helper, is better to do it in the manifest itself to have all the logic there.

It could look like an anti-pattern but it is because you need to see the logic of the error handling in the same manifest like we do in the `common-library`: https://github.com/newrelic/helm-charts/blob/ebf870a/library/common-library/templates/_license_secret.yaml.tpl